manifests/plugin.pp

```diff
@@ -194,9 +192,18 @@
       proxy_server    => $jenkins::proxy::url,
       cleanup         => false,
       extract         => false,
+      user            => $jenkins::user,
+      group           => $jenkins::group,
       require         => $plugindir,
       notify          => $notify,
     }
+
+    exec { "${plugin} permissions":
+      command   => "chown ${jenkins::user}:${jenkins::group} ${jenkins::plugin_dir}/${plugin}",
+      path      => '/usr/sbin:/usr/bin',
+      subscribe => Archive[$plugin],
+    }
```

Comment:
Can't this be done with a file resource?

You'd also think that this part takes care of it already:
https://github.com/voxpupuli/puppet-archive/blob/ac5a8a4e756499dfa315efe2a63abe6383bd00ca/lib/puppet_x/bodeco/archive.rb#L34-L57

Reply:
The exec is straight out of archive's readme fwiw since we don't crate the plugins dir ourselves.

Reply:
I've used the file resource in similar ways before. Not with archive, but where native type ends up creating a directory. As long as the dependency is there, it's fine to add one and avoids an exec (which is always a good thing to avoid when possible).

Reply:
so, the exec did resolve the issue I was seeing where permissions needed to be fixed. There seems to be a second layer to this though, and the file resource might not have been the root of why there was a second run needed - it seem that there might be an ordering issue as it kinda looks like `jenkins::config` is executing its ensure_resource on the credentials plugin (maybe others too) on the second run.

Reply:
There's a [fact](https://github.com/voxpupuli/puppet-jenkins/blob/master/lib/facter/jenkins.rb) that fails. Perhaps this part is broken in the container:
https://github.com/voxpupuli/puppet-jenkins/blob/15ffa7160bf92546c9246d4a7644457ed06732df/lib/puppet/jenkins.rb#L6

Reply:
ooooo that could do it... looking

Reply:
I may be having a memory lapse, but wouldn't the fact always fail on the first run as that user does not exist pre-manifest application when facts are evaluated?

Reply:
Oh yes, that's very likely. Perhaps it should have a fallback to `/var/lib/jenkins` which AFAIK is the default in packaging.